io-engine/src/grpc/v1/stats.rs

```diff
@@ -193,3 +267,23 @@ impl From<ExternalType<(String, BlockDeviceIoStats)>> for IoStats {
         }
     }
 }
+
+/// Returns IoStats for a given BlockDevice.
+async fn get_pool_stats(
+    name: String,
+    uuid: String,
+    bdev: UntypedBdev,
+) -> Result<IoStats, CoreError> {
+    let stats = bdev.stats_async().await?;
+    Ok(IoStats::from(ExternalType((name, uuid, stats))))
+}
+
+/// Returns IoStats for a given Nexus.
+async fn get_nexus_stats(
+    name: String,
+    uuid: String,
+    bdev: &Nexus<'_>,
```

Comment:
hmm this function is identical to the one above if you can pass the nexus as generic bdev somehow, and keep only one function.

Reply:
In the Nexus struct. Bdev is not a generic bdev. It returns Bdev<Nexus<`n> >. This is why we decided to return stats directly from Nexus impl. This method is to group the conversion also in same fn.